src/dynamixel_hardware_interface.cpp

```diff
@@ -144,24 +179,41 @@ namespace dynamixel
     **/
     void DynamixelHardwareInterface::write_joints()
     {
-        std::vector<int> command_int;
-        command_int.resize(_dynamixel_ids.size());
 
-        for(unsigned int i=0; i<_dynamixel_ids.size(); i++)
-        {
-            command_int[i] = (int)(_joint_commands[i]*2048/M_PI) + 2048;
+        for (unsigned int i = 0; i < _dynamixel_servos.size(); i++) {
+            std::map<long long int, std::string>::iterator dynamixel_iterator = _dynamixel_map.find(_dynamixel_servos[i]->id());
+            if (dynamixel_iterator == _dynamixel_map.end())
+                continue;
+            // Sending commands only when needed
+            if (std::abs(_joint_commands[i] - _prev_commands[i]) < std::numeric_limits<double>::epsilon())
+                continue;
+            _prev_commands[i] = _joint_commands[i];
+            try {
+                dynamixel::StatusPacket<dynamixel::protocols::Protocol1> status;
+
+                double goal_pos = _joint_commands[i];
+
+                std::map<long long int, double>::iterator dynamixel_corrections_iterator = _dynamixel_corrections.find(_dynamixel_servos[i]->id());
+                if (dynamixel_corrections_iterator != _dynamixel_corrections.end()) {
```

Comment:
> if (dynamixel_iterator == _dynamixel_map.end())
> +                continue;

You mean this one right?

This was a quick "workaround" (it's not exactly a workaround) for the following:

The `dynamixel::auto_detect` (look [here](https://github.com/resibots/dynamixel_control_hw/blob/dyn_v2/src/dynamixel_hardware_interface.cpp#L38)) returns all the available dynamixel found on the serial interface. But in ROS we may not want to control all of them (we read the joints we want from the yaml file). That's why we check. Right now, it's not very good performance-wise, as we search the map in every loop. On the other hand, this map never contains more than 20 elements, so searching it is almost free. We should look into this though.